draft-irtf-cfrg-vdaf.md

```diff
@@ -3153,107 +3162,106 @@ struct {
+field element. If the value is zero, then the measurement is deemed valid;
+otherwise, if the output is non-zero, then the measurement is deemed invalid.
 
-As an illustrative example, consider a validity circuit `C` that recognizes the
-set `L = set([0], [1])`. That is, `C` takes as input a length-1 vector `x` and
-returns 0 if `x[0]` is in `[0,2)` and outputs something else otherwise. This
-circuit can be expressed as the following degree-2 polynomial:
+For example, the simplest circuit specified in this document is the following
+({{prio3count}}):
 
 ~~~
-C(x) = (x[0] - 1) * x[0] = x[0]^2 - x[0]
+C(x) = x * (x-1)
 ~~~
 
-This polynomial recognizes `L` because `x[0]^2 = x[0]` is only true if `x[0] ==
-0` or `x[0] == 1`. Notice that the polynomial involves a non-affine operation,
-`x[0]^2`. In order to apply {{BBCGGI19}}, Theorem 4.3, the circuit needs to be
-rewritten in terms of a gadget that subsumes this non-affine operation. For
-example, the gadget might be multiplication:
+This circuit contains one negation gate (`-1`), one addition gate (`x + (-1)`),
+and one multiplication gate (`x * (x + (-1))`). Observe that `C(x) = 0` if and
+only if `x in range(2)`.
+
+Our goal is to allow each Aggregator, who holds a secret share of `x`, to
+correctly compute a secret share of `C(x)`. This allows the Aggregators to
+determine validity by combining their shares of the output.
+
+Suppose for a moment that the validity circuit `C` is affine, meaning its only
+operations are negation, addition, and multiplication-by-constant. (The circuit
+above is non-affine because it contains a multiplication gate with non-constant
+inputs.) Then each Aggregator can compute its share locally, since
 
 ~~~
-Mul(left, right) = left * right
+C(x_shares[0] + ... + x_shares[SHARES-1]) =
+    C(x_shares[0]) + ... + C(x_shares[SHARES-1])
 ~~~
 
-The validity circuit can then be rewritten in terms of `Mul` like so:
+(Note that, for this equality to hold, it is necessary to scale any constants
+in the circuit by `1/SHARES`.) However, this is not the case if `C` is
+multiplication gates with non-constant inputs. Thus or goal is to transform
+these multiplication gates into computations on secret shared data that each
+Aggregator can perform locally.
+
+The key idea is to have the prover construct a polynomial `p` such that `p(j)`
+is equal to the output of the `j`-th multiplication gate. Polynomial evaluation
+is fully linear, which means the coefficients of the polynomial can be secret
+shared in a way that allows each Aggregator to compute a share of `p(j)` for
+any `j`. These intermediate results can then be combined with the affine
+operations of the validity circuit to produce the final output.
+
+Applying this idea to the example circuit `C` above:
+
+1. The Client, given its measurement `x`, constructs the lowest degree
+   polynomial `p` for which `p(0) = x * (x - 1)`. It sends secret shares of `x`
+   and the coefficients of `p` to each of the Aggregators.
```

Comment:
This isn't quite right, because the wire polynomials evaluated at the first point produce blinding values. (and the gadget polynomial evaluated at the first point outputs some function of the blinding values) I think we should probably mention or allude to the blinding here, since it's necessary for zero knowledge, and it's not mentioned below.

Reply:
Nice catch :) Please take another look: I used `s` to denote the blinding factor, which is short for `seed`, the word we use in the actual specification.